src/components/shared/OTPInput/OTP/OTPhook.ts

```diff
@@ -0,0 +1,156 @@
+import { useRef, Dispatch, SetStateAction } from 'react';
+
+interface IOtpFunctionsProps {
+  length: number;
+  value: string;
+  onChange: Dispatch<SetStateAction<string>>;
+}
+
+const useOtpFunctions = ({ length, value, onChange }: IOtpFunctionsProps) => {
+  const inputRefs = useRef<HTMLInputElement[]>(new Array(length).fill(null));
+
+  const focusInput = (targetIndex: number): void => {
+    const targetInput = inputRefs.current[targetIndex];
+    targetInput.focus();
+  };
+
+  const selectInput = (targetIndex: number): void => {
+    const targetInput = inputRefs.current[targetIndex];
+    targetInput.select();
+  };
+
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+    currentIndex: number
+  ): void => {
+    switch (event.key) {
+      case ' ':
+        event.preventDefault();
+        break;
+
+      case 'ArrowLeft':
+        event.preventDefault();
+        if (currentIndex > 0) {
+          focusInput(currentIndex - 1);
+          selectInput(currentIndex - 1);
+        }
+        break;
+
+      case 'ArrowRight':
+        event.preventDefault();
+        if (currentIndex < length - 1) {
+          focusInput(currentIndex + 1);
+          selectInput(currentIndex + 1);
+        }
+        break;
+
+      case 'Delete':
+        event.preventDefault();
+        onChange((prevOtp) => {
+          const otp =
+            prevOtp.slice(0, currentIndex) + prevOtp.slice(currentIndex + 1);
+          return otp;
+        });
+        break;
+
+      case 'Backspace':
+        event.preventDefault();
+        if (currentIndex > 0) {
+          focusInput(currentIndex - 1);
+          selectInput(currentIndex - 1);
+        }
+
+        onChange((prevOtp) => {
+          const otp =
+            prevOtp.slice(0, currentIndex) + prevOtp.slice(currentIndex + 1);
+          return otp;
+        });
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    currentIndex: number
+  ): void => {
+    const currentValue = event.target.value;
+    let indexToEnter = 0;
+
+    while (indexToEnter <= currentIndex) {
+      if (
+        inputRefs.current[indexToEnter].value &&
+        indexToEnter < currentIndex
+      ) {
+        indexToEnter += 1;
+      } else {
+        break;
+      }
+    }
+    onChange((prev) => {
+      const otpArray = prev.split('');
```

Comment:
I assume some comments should be added, to shortly describe what is going on here

Reply:
Otp input is one parent component which has 6 small inputs. Parent component returns string like "123456", but inside it contains inputs values in an array like ["1", "2"...], so to make changes to the input value handler first splits the input value string to array of strings, makes changes and returns the joined string. The handlers work quite nice providing all functionality of otp: accept only one symbol, once one symbol is entered or deleted - the next input becomes focused, paste of all otp code.